simphony/io/file_lattice.py

```diff
@@ -0,0 +1,150 @@
+from simphony.io.indexed_data_container_table import IndexedDataContainerTable
+from simphony.io.data_container_description import NoUIDRecord
+from simphony.core.data_container import DataContainer
+
+import numpy as np
+
+
+class FileLattice(ABCLattice):
+    """
+    FileLattice object to use in file lattices.
+
+    Attributes
+    ----------
+    file : PyTables file
+        reference to PyTables file where the lattice is or will be located
+    name : str
+        name of the lattice
+    type : string
+        Bravais lattice type (should agree with the _base_vect below).
+    base_vect : D x float
+        defines a Bravais lattice (an alternative for primitive vectors).
+    size : D x size
+        number of lattice nodes (in the direction of each axis).
+    origin : D x float
+        origin of lattice
+    record : tables.IsDescription
+        A class that describes column types for PyTables table.
+
+    Returns:
+    ----------
+    FileLattice
+        The lattice newly added to the file or existing in the file.
+
+    """
+    def __init__(self, file, name, type, base_vect, size, origin,
+                 record=None):
+        self.name = name
+        self._type = type
+        self._base_vect = np.array(base_vect, dtype=np.float)
+        self._size = tuple(size)
+        self._origin = np.array(origin, dtype=np.float)
+        self._file = file
+        self._group = file.root.lattice
+        self._record = record
+
+        # If Lattice not in file, create a lattice
+        if self.name not in self._file.root.lattice:
+            if self._record is None:
+                # If record not specified use NoUIDRecord
+                self._record = NoUIDRecord
+            self._table = IndexedDataContainerTable(
+                self._group, name, self._record, np.prod(self._size))
+            for i in xrange(np.prod(self._size)):
+                self._table.append(DataContainer())
+
+            # Set table attributes
+            lattice = self._group._f_getChild(self.name)
+            lattice.attrs.type = self._type
+            lattice.attrs.base_vect = self._base_vect
+            lattice.attrs.size = self._size
+            lattice.attrs.origin = self._origin
+
+        # If Lattice already in file, return a reference
+        else:
+            self._table = IndexedDataContainerTable(self._group, self.name)
+            lattice = self._group._f_getChild(self.name)
+            self._type = lattice.attrs.type
+            self._base_vect = lattice.attrs.base_vect
+            self._size = lattice.attrs.size
+            self._origin = lattice.attrs.origin
+
+    def get_node(self, index):
+        """Get a copy of the node corresponding to the given index.
+
+        Parameters:
+        -----------
+        index : tuple of D x int (node index coordinate)
+
+        Returns:
+        -----------
+        A reference to a LatticeNode object
+
+        """
+        n = np.ravel_multi_index(index, self._size)
```

Comment:
I was not sure if these methods where doing the right thing. Thanks for looking up